### ISSUE TITLE
Upgrade netty-tcnative-boringssl-static to support uber-jar dependency

### ISSUE THREAD

**Issue by swegner** [GoogleCloudPlatform/cloud-bigtable-client#917]
**TL:DR:** [Apache Beam](https://github.com/apache/incubator-beam) would like to upgrade `cloud-bigtable-client`'s dependency on `netty-tcnative-boringssl-static` in order to link against the `netty-tcnative` "uber" jar artifact introduced in Fork16.

In order to use Bigtable from a Maven project, users must include `netty-tcnative-boringssl-static` and add the `os-maven-plugin` to bind to the platform-specific artifact ([docs](https://cloud.google.com/bigtable/docs/using-maven#encryption)). In [Apache Beam](https://github.com/apache/incubator-beam), we support Bigtable in our [GCP IO module](https://github.com/apache/incubator-beam/blob/master/sdks/java/io/google-cloud-platform/pom.xml), and thus `netty-tcnative-boringssl-static` becomes a transitive dependency. As a result, when a Beam user creates a project using GCP IO, they must also add `os-maven-plugin` to resolve the `netty-tcnative-boringssl-static` dependency or else Maven fails with a strange error. See: [[BEAM-445] Beam-examples-java build failed through local "mvn install"](https://issues.apache.org/jira/browse/BEAM-445).

We'd like to encapsulate the dependency on `netty-tcnative` so users can just add our Beam GCP IO dependency and everything else "just works". One solution would be to target the `netty-tcnative` ["uber" jar](http://netty.io/wiki/forked-tomcat-native.html#how-to-download-netty-tcnative-boringssl-uber-jar), which contains all platform-specific libraries. The uber jar was [first released with `netty-tcnative` Fork16](https://github.com/netty/netty/issues/5052#issuecomment-202469579), but [Bigtable depends on version Fork13](https://github.com/netty/netty/issues/5052#issuecomment-202469579).

1. Has there been any testing with `netty-tcnative` >= Fork16? Are these versions supported?
2. Could we upgrade the dependency version of `netty-tcnative` in Bigtable?

**Comment by swegner** [comment]
CC: @kennknowles @dhalperi

**Comment by sduskis** [comment]
Our next release, 0.9.1 will support the uberjar.  For more information, let's please talk offline.

**Comment by faizalkassamalisc** [comment]
has this been fixed?  I'm using 0.9.1 but seeing

```
DEBUG [15:25:16,948Z] (BigtableSession-startup-0)  io.netty.handler.ssl.OpenSsl: Failed to load netty-tcnative; OpenSslEngine will be unavailable. See http://netty.io/wiki/forked-tomcat-native.html for more information.
java.lang.UnsatisfiedLinkError: no netty-tcnative in java.library.path
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1864)
	at java.lang.Runtime.loadLibrary0(Runtime.java:870)
	at java.lang.System.loadLibrary(System.java:1122)
	at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:168)
	at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:58)
	at com.google.cloud.bigtable.grpc.BigtableSession.isAlpnProviderEnabled(BigtableSession.java:120)
	at com.google.cloud.bigtable.grpc.BigtableSession$1.run(BigtableSession.java:154)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

and 

```
        Caused by:
        java.lang.IllegalStateException: Could not initialize a BigtableClusterAdminClient

            Caused by:
            java.lang.IllegalStateException: Neither Jetty ALPN nor OpenSSL via netty-tcnative were properly configured.
```

**Comment by sduskis** [comment]
0.9.1 is out.  The instructions for maven are at: https://cloud.google.com/bigtable/docs/using-maven.

If that doesn't work, please email me and/or [this group](https://groups.google.com/forum/#!forum/google-cloud-bigtable-discuss) with your pom.xml.

**Comment by faizalkassamalisc** [comment]
I see from the link you sent that 0.9.1 is out for both bigtable-hbase-1.1 and bigtable-hbase-1.2.  Which should we be using.  This is a little confusing.

In any case I tried switching to bigtable-hbase-1.2 v0.9.1 and I'm getting the same error.

**Comment by sduskis** [comment]
Which version of tcnative are you using?

**Comment by faizalkassamalisc** [comment]
```
    compile "io.netty:netty-tcnative:1.1.33.Fork19"
    compile "io.netty:netty-tcnative-boringssl-static:1.1.33.Fork19"
```

**Comment by sduskis** [comment]
Hm... I never tried both netty-tcnatve and netty-tcnative-boringssl-static.  Can you please comment out the following:

`compile "io.netty:netty-tcnative:1.1.33.Fork19"`
